src/OpenLoco/Windows/ScenarioSelect.cpp

```diff
@@ -0,0 +1,507 @@
+        self->initScrollWidgets();
+
+        self->colours[0] = PaletteIndex::transparent;
+        self->colours[1] = PaletteIndex::index_0B;
+
+        self->var_846 = 0xFFFF;
+        self->var_85A = static_cast<int32_t>(0xFFFFFFFF);
+
+        initTabs(self);
+
+        // Select the last tab used, or the first available one.
+        uint8_t selectedTab = config::get().scenario_selected_tab;
+        if (self->widgets[widx::tab0 + selectedTab].type == widget_type::none)
+        {
+            selectedTab = 0;
+            for (int i = 0; i < 5; i++)
+            {
+                if (self->widgets[widx::tab0 + i].type == widget_type::none)
+                {
+                    selectedTab = i;
+                    break;
+                }
+            }
+        }
+
+        self->current_tab = selectedTab;
+
+        initList(self);
+
+        return self;
+    }
+
+    // 0x00443995
+    static void prepareDraw(window* self)
+    {
+        self->activated_widgets &= ~((1 << widx::tab0) | (1 << widx::tab1) | (1 << widx::tab2) | (1 << widx::tab3) | (1 << widx::tab4));
+        self->activated_widgets |= (1ULL << (self->current_tab + static_cast<uint8_t>(widx::tab0)));
+    }
+
+    // 0x004439AF
+    static void draw(window* self, Gfx::drawpixelinfo_t* dpi)
+    {
+        Gfx::drawRectInset(dpi, self->x, self->y + 20, self->width - 1, 1, self->colours[0], 0);
+
+        // Draw widgets.
+        self->draw(dpi);
+
+        static const string_id scenarioGroupIds[] = {
+            string_ids::scenario_group_beginner,
+            string_ids::scenario_group_easy,
+            string_ids::scenario_group_medium,
+            string_ids::scenario_group_challenging,
+            string_ids::scenario_group_expert,
+        };
+
+        // Draw tab captions.
+        for (int i = 0; i < 5; i++)
+        {
+            widget_t& widget = self->widgets[widx::tab0 + i];
+            if (widget.type == widget_type::none)
+                continue;
+
+            const auto offset = self->current_tab == i ? 1 : 0;
+            auto origin = Gfx::point_t(widget.mid_x() + self->x, widget.mid_y() + self->y - 3 - offset);
+            const string_id caption = scenarioGroupIds[i];
+
+            Gfx::drawStringCentredWrapped(dpi, &origin, 87, Colour::black, string_ids::wcolour2_stringid, &caption);
+        }
+
+        // Scenario selected?
+        if (self->info == 0 || self->info == 0xFFFFFFFF)
+            return;
+
+        using namespace scenariomgr;
+        auto scenarioInfo = reinterpret_cast<ScenarioIndexEntry*>(self->info);
+
+        // Check if current currency object needs to be changed.
+        // TODO: this segfaults.
+        registers regs;
+        regs.ebp = scenarioInfo->currentObjectPtr;
+        printf("Pointer: %p\n", (void*)scenarioInfo->currentObjectPtr);
+        bool notLoaded = call(0x004720EB, regs) & (X86_FLAG_SIGN << 8);
```

Comment:
This segfaults, but I'm not sure where the fault is. The pointer value I'm getting in stdout is `0x118d0282`, which seems a bit high.